src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -802,6 +804,12 @@ protected void onResume() {
         super.onResume();
         Log_OC.e(TAG, "onResume() start");
         
+        if (PinCheck.checkIfPinEntry()){
+            Intent i = new Intent(MainApp.getAppContext(), PinCodeActivity.class);
```

Comment:
Since this is duplicated in onCreate(...) and in onResume(), the code to check the PIN code is being called twice when the Activity is initialized. Now it's working because the timeout for the PIN code is 10 seconds, and it's impossible so much time goes from onCreate() to onResume() execution; but, if there were no timeout, then we would ask the PIN code twice to the user. It is risky, thinking about possible future changes.

Besides, when the user comes back from a modal dialog, onResume() is also called. I didn't test it yet, but I bet that if we click the 'upload' icon, wait for 10 seconds doing nothing, and the cancel, we will be requested for the PIN. All of it without exiting the app.

Probably the best place to do the check is onStart()  (onCreate() is not good enough); it is called every time the activity is becoming visible, but not after modal dialogs partially hiding the app disappear.